src/opm/output/eclipse/EclipseIO.cpp

```diff
@@ -208,6 +211,16 @@ void EclipseIO::writeTimeStep(const SummaryState& st,
         this->impl->summary.write();
     }
 
+    auto length { this->impl->schedule.posixEndTime() - this->impl->schedule.posixStartTime() } ;
+    auto timeStep { this->impl->schedule.stepLength(report_step) } ;
+    bool final_step { secs_elapsed >= length - timeStep } ;
```

Comment:
This can just be: `bool final_step { report_step == this->impl.schedule.size() };`